src/main/java/org/apache/sling/feature/scanner/impl/FelixFrameworkScanner.java

```diff
@@ -143,23 +144,31 @@ public Manifest getManifest() {
     Map<String,String> getFrameworkProperties(final Map<String,String> appProps, final File framework)
     throws IOException {
         final Map<String, Properties> propsMap = new HashMap<>();
-        try (final ZipInputStream zis = new ZipInputStream(new FileInputStream(framework)) ) {
-            boolean done = false;
-            while ( !done ) {
-                final ZipEntry entry = zis.getNextEntry();
-                if ( entry == null ) {
-                    done = true;
-                } else {
+        ZipFile zipFile = null;
```

Comment:
Same as above - try() blocks and only open streams if you need them.

Reply:
@karlpauls Just pushed the changes as suggested. Thank you for reviewing this.